src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -140,7 +145,10 @@ public String deploy(AppDeploymentRequest request) {
 				.push(PushApplicationRequest.builder()
 					.name(name)
 					.application(request.getResource().getFile().toPath())
-					.domain(connectionProperties.getDomain())
+					.domain(domain(request))
+					.host(host(request))
+					.routePath(routePath(request))
+					.noRoute(toggleNoRoute(request))
```

Comment:
Should we cross check `routePath` against `noRoute` as both these properties seem mutually exclusive?